vtysh/vtysh.c

```diff
@@ -4721,6 +4721,25 @@ static int vtysh_connect(struct vtysh_client *vclient)
 		close(sock);
 		return -1;
 	}
+
+	/* Setting the RCV buffer size to 16MB to increase the buffer size
+	 * the socket can hold after receving from other process
+	 */
+
+	uint32_t rcvbufsize = BUFFER_SIZE_DEFAULT * BUFFER_SIZE_DEFAULT;
+
+	ret = setsockopt(sock, SOL_SOCKET, SO_RCVBUF, (char *)&rcvbufsize,
+			 sizeof(rcvbufsize));
+	if (ret < 0) {
+#ifdef DEBUG
```

Comment:
Why do we need this?

Reply:
When we flush intermittently, `vtysh_flush_intermediate` is called multiple times because the entire buffer is not flushed at once (only 131KB is flushed in the current code). These multiple calls are scheduled because there is no space in the send buffer to place the data in the socket. This blocks socket calls from other terminals, meaning if we try to run another command in a different terminal on the same node, it is blocked for a while because its connection request is not honored due to the numerous scheduled `vtysh_flush_intermediate` calls. By increasing the send/receive buffer size, more data can be placed on the socket, which will prevent multiple `vtysh_flush_intermediate` calls in the mean time from blocking other terminals.

Reply:
I meant `ifdef DEBUG`.

Reply:
I think we are just following how fprintf is guarded by ifdef in the function

Reply:
I'm with Donatas here - let's log it, or debug it, but ... there's not much value in using stderr?